hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/ssl/DelegatingSSLSocketFactory.java

```diff
@@ -167,6 +167,15 @@ private void initializeSSLContext(SSLChannelMode preferredChannelMode)
         LOG.debug("Failed to load OpenSSL. Falling back to the JSSE default.");
         ctx = SSLContext.getDefault();
         channelMode = SSLChannelMode.Default_JSSE;
+      } catch (Exception e) {
```

Comment:
should we remove the `catch` block for the `NoSuchAlgorithmException` and just let all exceptions be handled by the same block? these changes look useful for the `NoSuchAlgorithmException` as well